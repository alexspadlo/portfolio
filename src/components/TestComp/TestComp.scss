@font-face {
  font-family: MyRalewayBold;
  src: url("./ralewayBold.woff2") format("woff2");
}

.batata {
  background-color: purple;
}

.some {
  color: white;
  font-family: MyRalewayBold, sans-serif;
}
