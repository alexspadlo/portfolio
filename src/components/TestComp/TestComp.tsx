import React from 'react'
// import testVid from './drop_1.mp4'
import s from './TestComp.scss'
import '@google/model-viewer'
import batata from '../../static/images/avatar.glb'

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'model-viewer': MyElementAttributes;
        }
        interface MyElementAttributes {
            src: string;
        }
    }
}

const TestComp: React.FC = () => {
    return (
        <section>
            <div className='container'>
                <div className='row'>
                    <div className='col-12'>
                        <p className={`${s.batata} h3 py-4 text-center ${s.some}`}>Webpack Typescript Test</p>
                        <model-viewer auto-rotate camera-controls src={batata}></model-viewer>
                        {/* <video className='img-fluid w-100' controls>
                            <source src={testVid} />
                        </video> */}
                    </div>
                </div>
            </div>
        </section>

    )
}

export default TestComp