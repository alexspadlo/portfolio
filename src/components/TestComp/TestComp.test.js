import React from 'react'
import renderer from 'react-test-renderer'
import TestComp from './TestComp'
jest.mock('@google/model-viewer')

describe('Test on <TestComp /> component', () => {
    test('Should render without the props', () => {
        const componentNoprops = renderer.create(<TestComp />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
})
