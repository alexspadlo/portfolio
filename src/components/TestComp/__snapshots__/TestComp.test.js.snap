// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TestComp /> component Should render without the props 1`] = `
<section>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <p
          className="batata h3 py-4 text-center some"
        >
          Webpack Typescript Test
        </p>
        <model-viewer
          auto-rotate={true}
          camera-controls={true}
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</section>
`;
